corehq/apps/geospatial/templates/gps_capture.html

```diff
@@ -79,20 +79,37 @@ <h3 class="panel-title">
                             </a>
                         </td>
                         <td>
-                            <input class="form-control"
-                                   type="number"
-                                   step="any"
-                                   name="latInput"
-                                   placeholder="{% trans 'Enter latitude...' %}"
-                                   data-bind="value: lat, event: {change: onValueChanged}" />
+                            <div data-bind="css: {'has-error': !isLatValid()}">
+                                <input class="form-control"
+                                       type="number"
+                                       step="any"
+                                       name="latInput"
+                                       placeholder="{% trans 'Enter latitude...' %}"
+                                       data-bind="value: lat, event: {change: onValueChanged}" />
```

Comment:
The event gets fired as soon as the user leaves the text box. This won't allow users to save with invaldiated data however, so if they click on the save button with invalid input, then the save button will be disabled before the save click handler fires.

Reply:
> The event gets fired as soon as the user leaves the text box.

Cool, just wanted to confirm that the user wont see the error message while they start typing.

> if they click on the save button with invalid input

I assume the button would be disabled till the input box has valid inputs?

Reply:
> I assume the button would be disabled till the input box has valid inputs?

That is correct, but it might be that they initially have valid input so the save button gets enabled. If they then enter invalid input and attempt to click on the save button, then the button will become disabled again before their click registers.